edx_exams/apps/lti/tests/test_views.py

```diff
@@ -193,6 +205,171 @@ def test_acs_no_attempt_found(self,
+        ('5', '1.0', 'error'),
+        ('6', '1.0', 'error'),
+        ('7', '1.0', 'error'),
+        ('8', '1.0', 'error'),
+        ('9', '1.0', 'error'),
+        ('12', '1.0', 'error'),
+        ('13', '1.0', 'error'),
+        ('14', '1.0', 'error'),
+        ('15', '1.0', 'error'),
+        ('16', '1.0', 'error'),
+        ('21', '1.0', 'error'),
+        ('24', '1.0', 'error'),
+        ('25', '1.0', 'error'),
+        # Testing the incident severity
+        ('1', '0.3', 'second_review_required'),
+        ('1', '0.26', 'second_review_required'),
+        ('1', '0.25', 'verified'),
+        ('1', '0.1', 'verified'),
+        # NOTE: I wanted to include a test where we pass in 0 (as an int).
+        # However, a unit test with integer 0 as incident_severity breaks this test for some weird reason.
+        # The 'incident_severity' variable turns into 'incident_incident_severity' in the data passed to views.py.
+        # Passing in 0 as an integer value works when using Postman, so I have ignored this case for now.
+    )
+    @ ddt.unpack
+    @ patch.object(Lti1p3ApiAuthentication, 'authenticate', return_value=(AnonymousUser(), None))
+    @ patch('edx_exams.apps.lti.views.LtiProctoringAcsPermissions.has_permission')
+    @ patch('edx_exams.apps.lti.views.get_attempt_for_user_with_attempt_number_and_resource_id')
+    def test_acs_terminate(self,
+                           reason_code,
+                           incident_severity,
+                           expected_attempt_status,
+                           mock_get_attempt,
+                           mock_permissions,
+                           mock_authentication):  # pylint: disable=unused-argument
+        """
+        Test that the terminate action changes the exam attempt status as expected
+        based on the 'reason_code' and 'incident_severity'.
+        """
+        self.attempt.status = ExamAttemptStatus.submitted
+        self.attempt.save()
+
+        mock_get_attempt.return_value = self.attempt
+        mock_permissions.return_value = True
+
+        token = self.make_access_token('https://purl.imsglobal.org/spec/lti-ap/scope/control.all')
+
+        request_body = self.create_request_body(
+            self.attempt.attempt_number,
+            'terminate',
+            reason_code,
+            incident_severity
+        )
+
+        # Even though the client.post function below uses json.dumps to serialize the request as json,
+        # The json serialization needs to happen before the request for an unknown reason
+        request_body = json.dumps(request_body)
+        self.client.post(self.url, data=request_body, content_type='application/json',
+                         HTTP_AUTHORIZATION='Bearer {}'.format(token))
+        self.attempt.refresh_from_db()
+
+        self.assertEqual(self.attempt.status, expected_attempt_status)
+
+        # Assure an entry was added to the ACResult model
+        data = AssessmentControlResult.objects.all()
```

Comment:
since we know the attempt id already an improvement to this test would to be to query for that result specifically.